pkg/models/message.go

```diff
@@ -16,6 +16,7 @@ import (
 type Message struct {
 	PartitionKey string
 	Data         []byte
+	HTTPHeaders  map[string][]string
```

Comment:
How come it's a map of slice[string] instead of a map of string?

Reply:
This type matches [the Header type](https://pkg.go.dev/net/http#Header) of the net/http package. One advantage is the ability to add many values to a specific header. This is not necessary for the gtmss preview header, but i went for the more generic case. Thoughts?

Reply:
Chatted about this on a call - have explained what I think but it's up to you - I don't think it makes a huge difference. :)